Intersect.Server/Entities/Player.cs

```diff
@@ -1625,31 +1625,36 @@ public bool CanGiveItem(Item item)
                 if (item.Descriptor.IsStackable)
                 {
                     // Does the user have this item already?
-                    var existingSlot = FindInventoryItemSlot(item.ItemId);
-                    if (existingSlot != null)
-                    {
-                        // Can we blindly add more to this stack?
-                        var untilFull = item.Descriptor.MaxInventoryStack - existingSlot.Quantity;
-                        if (untilFull >= item.Quantity)
-                        {
-                            return true;
-                        }
+                    var itemSlots = FindInventoryItemSlots(item.ItemId);
+                    var slotsRequired = Math.Ceiling((double)item.Quantity / item.Descriptor.MaxInventoryStack);
```

Comment:
How come this is cast to a double?
I see it used to be before as well but I'm curious what the reason was. lol

Reply:
But does math.ceiling make an actual difference here? Wouldn't an integer already round by default?  Curious if that would just work as well though might need some testing.

Reply:
Integer division _rounds_, but we need to get the ceiling, else we could lie about the amount of slots we need.